tests/integration/test_diff.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2022 NVIDIA Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import numpy as np
+import pytest
+from utils.comparisons import allclose
+
+import cunumeric as cn
+
+
+@pytest.mark.parametrize("n", [1, 2, 3])
+def test_diff(n):
+    num = np.random.random(100)
+    cun = cn.array(num)
+
+    res_num = np.diff(num, n=n)
+    res_cn = cn.diff(cun, n=n)
+
+    assert allclose(res_num, res_cn)
+
```

Comment:
empty edge case should get a test

Reply:
Also the case where `n` is greater than the length of the input

Reply:
And also need multi-dimensional / `axis` arg coverage

Reply:
updated